onmt/data/SampledDataset.lua

```diff
@@ -0,0 +1,224 @@
+--[[ Data management and batch creation. Handles data created by `preprocess.lua`. ]]
+require 'torchx'
+
+local SampledDataset = torch.class("SampledDataset")
+
+--[[ Initialize a data object given aligned tables of IntTensors `srcData`
+  and `tgtData`.
+--]]
+function SampledDataset:__init(srcData, tgtData, samplingSize, sample_w_ppl, sample_w_ppl_init, sample_w_ppl_max)
+
+  self.src = srcData.words
```

Comment:
Retrieving main feature from srcData and tgtData is now changed to "srcData.words or srcData.vectors" and  "tgtData.words or tgtData.vectors"; I hope this is enough. (08ad2ed)